### ISSUE TITLE
XGboost estimator will throw error when multiple evaluation metrics are given

### ISSUE THREAD

**Issue by everglory99** [instacart/lore#123]
In [xgboost scikit-learn fit API](https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBRegressor.fit), the argument `eval_metric` can be a string for single metric or a list of string for multiple metrics. However, if we provide a list to the `eval_metric`, it will break the `fit` method in `lore.estimators.xgboost.base`. Specifically, in [this](https://github.com/instacart/lore/blob/master/lore/estimators/xgboost.py#L70) and [that](https://github.com/instacart/lore/blob/master/lore/estimators/xgboost.py#L74). Can we support multiple eval metrics? I can submit a PR to fix this. Thanks!

**Comment by montanalow** [comment]
A PR would be much appreciated, including a test case that demonstrates the functionality.
